cmd/multitenant-autoscaler/main.go

```diff
@@ -79,12 +84,22 @@ func main() {
 		logger.Fatal("Error building kubernetes clientset.", zap.Error(err))
 	}
 
+	rm := restmapper.NewDeferredDiscoveryRESTMapper(cached.NewMemCacheClient(kubeClientSet.Discovery()))
```

Comment:
Lines 87-92 should be extracted to a suitably named function to improve readability.

Reply:
I factored this in a way that I could clearly link each half to the portion of kubernetes (HPA) logic it was based on, the second half ended up being a single line, so I left it inline.